include/bcrypt/BCrypt.hpp

```diff
@@ -1,26 +1,32 @@
-#ifndef __BCRYPT__
-#define __BCRYPT__
-
-#include "bcrypt.h"
-#include <string>
-#include <stdexcept>
-
-class BCrypt {
-public:
-    static std::string generateHash(const std::string & password, int workload = 12){
-        char salt[BCRYPT_HASHSIZE];
-        char hash[BCRYPT_HASHSIZE];
-        int ret;
-        ret = bcrypt_gensalt(workload, salt);
-        if(ret != 0)throw std::runtime_error{"bcrypt: can not generate salt"};
-        ret = bcrypt_hashpw(password.c_str(), salt, hash);
-        if(ret != 0)throw std::runtime_error{"bcrypt: can not generate hash"};
-        return std::string{hash};
-    }
-
-    static bool validatePassword(const std::string & password, const std::string & hash){
-        return (bcrypt_checkpw(password.c_str(), hash.c_str()) == 0);
-    }
-};
-
-#endif // __BCRYPT__
+#ifndef __BCRYPT__
+#define __BCRYPT__
+
+#ifdef _WIN32
+#include "winbcrypt.h"

```

Comment:
I don't get why we need to include a file which is really the same apart from two different includes. Couldn't we just add them in a little Windows include guard without creating a new file?

Reply:
Yes possibly, I started out with a new file because I thought it was going to be a lot more involved, started down the track of the windows BCrypt but sadly that is not BCrypt as we know but Microsofts "Best Crypt", I ended up being able to re-use existing code so yeah I think it could go in one file.